siege_utilities/survey/significance.py

```
@@ -0,0 +1,129 @@
+        df = df.drop(index=["Total"])
+
+    if df.empty or df.shape[0] < 2 or df.shape[1] < 2:
+        chain.chi_square_significant = False  # type: ignore[attr-defined]
+        chain.chi_square_p = None             # type: ignore[attr-defined]
+        return chain
+
+    try:
+        result = chi_square_test(df)
+        chain.chi_square_significant = result.p_value < alpha  # type: ignore[attr-defined]
+        chain.chi_square_p = result.p_value                    # type: ignore[attr-defined]
+    except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)